### PR TITLE
Ma 380 payload exception url encoding

### PR DESCRIPTION
**Pull request by PurushYM**
- Throwing payload exception on encoding failed

### CODE REVIEW

**Review by sukrit-ym — APPROVED** [review]

**Comment by kaunteya** [comment]
@PurushYM please dont merge this PR until iOS part is implemented

File: YMChat/src/main/java/com/yellowmessenger/ymchat/YMChat.java [hunk 1]
```diff
@@ -38,22 +43,36 @@ public void onEventFromBot(BotEventListener listener) {
         this.listener = listener;
     }
 
-    public void startChatbot(Context context) {
+
+    public void startChatbot(@NonNull Context context) throws Exception {
```
Comment (PriyankUpadhyay) [inline YMChat/src/main/java/com/yellowmessenger/ymchat/YMChat.java:47]: We can move all the validation checks to a separate private method.

Comment (sukrit-ym) [reply YMChat/src/main/java/com/yellowmessenger/ymchat/YMChat.java:47]: Lets keep validation here and move the start activity logic in private method.

Comment (sukrit-ym) [reply YMChat/src/main/java/com/yellowmessenger/ymchat/YMChat.java:46]: done

File: README.md [hunk 2]
```diff
@@ -71,7 +76,7 @@ ymChat.config.enableSpeech = true
 ```
 
 ### Payload
-Additional payload can be added in the form of key value pair, which is then passed to the bot
+Additional payload can be added in the form of key value pair, which is then passed to the bot. The value of payload can be either Primitive type or Map<String,String>
```
Comment (PriyankUpadhyay) [inline README.md:79]: Are we only supporting `Map<String,String>`

**Review by PriyankUpadhyay — APPROVED** [review]

**Pull request merged**